### ISSUE TITLE
minimum Ruby 2.3.1 is required

### ISSUE THREAD

**Issue by mmaauu** [Shopify/shopify_app#465]
Hi! I've being using the shopify_app gem, and today I've tried to generate a new Rails project with the gem, but I'm getting this error once I tried to generate shopify_app with api_key and secret:
( until yesterday it was working well) This is what I get, can you help me with this issue? 

Thanks in advance!

``
$ rails g shopify_app --api_key XXXXXXXXXXXXXXXXXXXX --secret XXXXXXXXXXXXXXXXXXXX
    generate  shopify_app:install --api_key XXXXXXXXXXXXXXXXXXXX --secret XXXXXXXXXXXXXXXXXXXX
      create  config/initializers/shopify_app.rb
      create  config/initializers/omniauth.rb
      insert  config/initializers/omniauth.rb
      create  app/views/layouts/embedded_app.html.erb
      create  app/views/layouts/_flash_messages.html.erb
       route  mount ShopifyApp::Engine, at: '/'
    generate  shopify_app:shop_model
c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:287:in `load':
c:/sites/XXXXXXXXX/config/initializers/omniauth.rb:9: syntax error, unexpected '.' (SyntaxError)...on['shopify.omniauth_params']&.with_indifferent_access...                               ^
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:287:in `block in load'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:259:in `load_dependency'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:287:in `load'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:648:in `block in load_config_initializer'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/notifications.rb:166:in `instrument'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:647:in `load_config_initializer'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:611:in `each'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:611:in `block in <class:Engine>'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:30:in `instance_exec'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:30:in `run'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:55:in `block in run_initializers'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:420:in `block (2 levels) in each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:419:in `block in each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:44:in `each'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:44:in `tsort_each_child'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:413:in `call'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:413:in `each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:345:in `each'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:345:in `call'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:54:in `run_initializers'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/application.rb:352:in `initialize!'
        from c:/sites/XXXXXXXXXXXXX/config/environment.rb:5:in `<top (required)>'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/application.rb:328:in `require'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/application.rb:328:in `require_environment!'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:157:in `require_application_and_environment!'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:143:in `generate_or_destroy'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:60:in `generate'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands.rb:18:in `<top (required)>'
        from bin/rails:4:in `require'
        from bin/rails:4:in `<main>'
    generate  shopify_app:home_controller
c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:287:in `load':
c:/sites/XXXXXXXXXX/config/initializers/omniauth.rb:9: syntax error, unexpected '.' (SyntaxError)...on['shopify.omniauth_params']&.with_indifferent_access...                               ^
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:287:in `block in load'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:259:in `load_dependency'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/dependencies.rb:287:in `load'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:648:in `block in load_config_initializer'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-5.0.5/lib/active_support/notifications.rb:166:in `instrument'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:647:in `load_config_initializer'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:611:in `each'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/engine.rb:611:in `block in <class:Engine>'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:30:in `instance_exec'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:30:in `run'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:55:in `block in run_initializers'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:420:in `block (2 levels) in each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:419:in `block in each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:44:in `each'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:44:in `tsort_each_child'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:413:in `call'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:413:in `each_strongly_connected_component_from'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:345:in `each'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:345:in `call'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
        from c:/Rails/Ruby2.2.0/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/initializable.rb:54:in `run_initializers'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/application.rb:352:in `initialize!'
        from c:/sites/XXXXXXXX/config/environment.rb:5:in `<top (required)>'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/application.rb:328:in `require'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/application.rb:328:in `require_environment!'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:157:in `require_application_and_environment!'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:143:in `generate_or_destroy'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:60:in `generate'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
        from c:/Rails/Ruby2.2.0/lib/ruby/gems/2.2.0/gems/railties-5.0.5/lib/rails/commands.rb:18:in `<top (required)>'
        from bin/rails:4:in `require'
        from bin/rails:4:in `<main>'
``

**Comment by kevinhughes27** [comment]
Woops! Looks like we added code that requires ruby 2.3.1 now. Can you
update your ruby version and try again?

**Comment by mmaauu** [comment]
Thank you @kevinhughes27, it worked! Thanks a lot!

**Comment by kevinhughes27** [comment]
Happy to help. If you have a second could you look into updating the gemspec and readme with minimum ruby versions?

**Comment by mmaauu** [comment]
No problem, could  you please guide me? Thanks!

**Comment by kevinhughes27** [comment]
I think we just need to update this line and possibly the CI config: https://github.com/Shopify/shopify_app/blob/master/shopify_app.gemspec#L11

**Comment by mmaauu** [comment]
I don't have this line of code in the shopify_app.gemspec, the file looks like this:

```
$LOAD_PATH.push File.expand_path('../lib', __FILE__)
require "shopify_app/version"

Gem::Specification.new do |s|
  s.name        = "shopify_app"
  s.version     = ShopifyApp::VERSION
  s.platform    = Gem::Platform::RUBY
  s.author      = "Shopify"
  s.summary     = %q{This gem is used to get quickly started with the Shopify API}

  s.add_dependency('rails', '>= 3.1', '< 5.0')

  s.add_runtime_dependency('shopify_api', '~> 4.0.2')
  s.add_runtime_dependency('omniauth-shopify-oauth2', '~> 1.1.10')

  s.add_development_dependency('rake')
  s.add_development_dependency('byebug')
  s.add_development_dependency('sqlite3')
  s.add_development_dependency('minitest')
  s.add_development_dependency('mocha')

  s.files         = `git ls-files`.split("\n")
  s.test_files    = `git ls-files -- {test}/*`.split("\n")
  s.require_paths = ["lib"]
end
```

**Comment by kevinhughes27** [comment]
I linked to the code on master so you need to pull down the latest code

**Comment by vfonic** [comment]
@kevinhughes27 seems like `closes #493 #465` syntax doesn't close both issues.

**Comment by kevinhughes27** [comment]
woops. Thanks!
